public/app/AppWrapper.tsx

```diff
@@ -144,3 +131,93 @@ export class AppWrapper extends Component<AppWrapperProps, AppWrapperState> {
     );
   }
 }
+
+function RouterTree(props: { routes?: JSX.Element | false }) {
+  return (
+    <Router history={locationService.getHistory()}>
+      <LocationServiceProvider service={locationService}>
+        <CompatRouter>
+          <ModalsContextProvider>
+            <GlobalStyles />
+            <AppChrome>
+              <AngularRoot />
+              <AppNotificationList />
+              <Stack gap={0} grow={1} direction="column">
+                {pageBanners.map((Banner, index) => (
+                  <Banner key={index.toString()} />
+                ))}
+                {props.routes}
+              </Stack>
+              {bodyRenderHooks.map((Hook, index) => (
+                <Hook key={index.toString()} />
+              ))}
+            </AppChrome>
+            <LiveConnectionWarning />
+            <ModalRoot />
+            <PortalContainer />
+          </ModalsContextProvider>
+        </CompatRouter>
+      </LocationServiceProvider>
+    </Router>
+  );
+}
+
+/**
+ * Renders both the main app tree and a secondary sidecar app tree to show 2 apps at the same time in a resizable split
+ * view.
+ * @param props
+ * @constructor
+ */
+function ExperimentalSplitPaneTree(props: { routes?: JSX.Element | false }) {
+  const appId = useSelector((state) => state.appSidecar.appId);
+  const dispatch = useDispatch();
+
+  const memoryLocationService = new HistoryWrapper(H.createMemoryHistory({ initialEntries: ['/'] }));
+
+  return (
+    <SplitPaneWrapper
+      splitOrientation="vertical"
+      paneSize={0.25}
+      minSize={200}
+      maxSize={200 * -1}
+      primary="second"
+      splitVisible={!!appId}
+      paneStyle={{ overflow: 'auto', display: 'flex', flexDirection: 'column' }}
+    >
+      <RouterTree routes={props.routes} />
```

Comment:
Feel very apprehensive about wrapping the whole page/app chrome in a spitter, which will break the body scrolling feature (still behind feature toggle but will soon be rolled out).   

Can't the sidecar app live inside a drawer like component (that can push content on larger screens but act as an overlay on smaller laptop screens) (and be rendered inside the Page layout), instead of outside the App 

@joshhunt @ashharrison90

Reply:
yeah `bodyScrolling` is being rolled out basically ~now

combination of drawer + `position: fixed;` panel is one option (similar to how the docked nav menu works). one complication here is the ability to resize the panel, which will mean we'll need to dynamically adjust padding on the app. think that's all doable, just a bit fiddly 🤔 

another option could be to only wrap the app when a sidecar app is opened. that leaves regular behaviour untouched and makes having a sidecar app open a special case, which it kind of is.

Reply:
Didn't check the `bodyScrolling` feature flag so have to do that first, right now didn't see any issues with scrolling.

> Can't the sidecar app live inside a drawer like component (that can push content on larger screens but act as an overlay on smaller laptop screens)

So if it's inside the drawer but the drawer pushes content, what is the difference between it and the split view? Why would it work better?

Or is that just for the possibility of having 2 modes one that acts as a split view vs a drawer on a smaller screen? Personally not sure if that would create too much of a difference in possible interactions, like again the drag and drop then would be impossible or awkward to work with. I think adjusting the design of the app to be responsive (considering the plan here is to make this all opt in) would be easier than making the UX work both in overlay drawer and a split view but that is mostly my gut feeling. We could also consider just not allow the side view at all on small screens, like the API that opens an app could just navigate to the app root instead in that case and it would basically work "the old way".

Reply:
@aocenas these will be subtle issues. easiest way to tell if the page is scrolling the body correctly is load a page that overflows, and try to arrow key up/down (without clicking the page first) to scroll the page. with these changes, it definitely won't work. there are other reasons we want `bodyScrolling` (better full page screenshots, url bar hiding correctly on mobile browsers, etc), but this is the easiest way to "test" it. 

the difference between a `position: fixed;` panel and the split view is that it won't break body scrolling 😄 look at the docked navigation menu for an example. that is a separate scrollable panel, but you can still arrow key up/down to scroll the main view. 

we might not have to go as far as that tho. i think
> another option could be to only wrap the app when a sidecar app is opened. that leaves regular behaviour untouched and makes having a sidecar app open a special case, which it kind of is.

would probably be enough

Reply:
So I tried the useSplitter hook as @torkelo suggested and it has the benefit of being able to control how things are wrapped. This means I can wrap in the way where without the second app there should be minimal difference so it does not interfere with the body scroll and at the same time does not cause remounting of the main app. The cost is two div wrappers that need to be styled in way that they behave the same as their parent.

I also had to add a switch to the hook because by default it sort of disallows overflow with `min-width: min-content` which in this case is a bit problematic.


https://github.com/user-attachments/assets/67122e03-4758-4f6d-9067-13068a518d4d

So this technically works well I think. I need to clean up the code still but please let me know if this approach makes sense.

Also keep in mind yeah this does not look that amazing right now and it's just a way to get the building blocks in main so we can start building the query library and see how it works together. At that point, we can still reevaluate whether resizing split makes the most sense but with a plugin actually using this and easy to test I think it will be much easier to talk about and I don't think changing this to some drawer or something else would add that much additional work later on.